src/voku/helper/AntiXSS.php

```diff
@@ -775,6 +775,19 @@ private function _do_never_allowed(string $str)
     private function _do_never_allowed_afterwards(string $str)
     {
         if (\stripos($str, 'on') !== false) {
+            $never_allowed_and_likely_contained = [];
+            $thirdChar = '';
+            foreach ($this->_never_allowed_on_events_afterwards as $event) {
```

Comment:
with this additional loop we reduce the pretty big list in 

https://github.com/voku/anti-xss/blob/ff6e54f4a98ad1cd28f8b4a0f3c3f92f3c421f0a/src/voku/helper/AntiXSS.php#L89-L306

which speeds up the process later on.

Reply:
in case you want a even more general solution, one could also think about a `Trie` datastructure - like symfony uses it for its routing component.

the problem we solve in this method is similar to matching a url against a set of routing-rules.